gobblin-config-management/src/main/java/gobblin/config/configstore/VersionComparator.java

```diff
@@ -0,0 +1,8 @@
+package gobblin.config.configstore;
+
+import java.util.Collection;
+
+public interface VersionComparator<V> {
```

Comment:
Could this just be an extension of Comparator<String>, and then we can just use java's max? No need for `getCurrentVersion`.